packages/grafana-ui/src/types/datasource.ts

```diff
@@ -208,6 +209,8 @@ export abstract class DataSourceApi<
    */
   abstract query(request: DataQueryRequest<TQuery>, observer?: DataStreamObserver): Promise<DataQueryResponse>;
 
+  observe?(request: DataQueryRequest<TQuery>): Observable<DataQueryResponsePacket>;
```

Comment:
should we remove `, observer?: DataStreamObserver` from the above function now or later?  I think we should go ahead and do it in this PR since there are only 3 builtin sources that use it and each one will help us test/improve the endpoint

Reply:
yes, I think we should.

Reply:
maybe even try to combine these 2 functions.

Reply:
> After we have refactored away our use of the old observer callback. Should we switch back to query method that can return Promise or Observable (like the old interface could). A bit messy now with having two interfaces, which should be abstract? which should plugin authors implement?

With this question in mind, would it make sense to say query is deprecated, don't use it for new code? Converting the old promise to observable should be easy `return from(promise)` and maybe we can create a helper that can convert `DataQueryResponse` to `DataQueryResponsePacket` for the simple use case of having only one. Seems like that would make for less ambiguity in what to implement (even though can seem more complex to do that for new devs due to the Observable in there).

Reply:
Ah I see I was a bit hasty with the comment I see you already do the wrapping in the `wrapOldQueryMethod`.

Reply:
<img width="925" alt="Screen Shot 2019-09-06 at 13 57 55" src="https://user-images.githubusercontent.com/10999/64426497-f1bb9480-d0ae-11e9-9a7c-b2fe3cfcd3dc.png">

I have unified the query method to a single method that can either return Promise<DataQueryResponse> or Observable<DataQueryResponse> 

that way it's easier for data sources to use whatever fits best and it's a single method, not two different optionally implemented ones that we have to check exists.  and from() can easily handle the Promise response and transform it into an Observable.

I also removed the DataQueryResponsePacket and added the optional key property to DataQueryResponse so we can have a single response type.